pkg/detectors/azure_entra/serviceprincipal/v2/spv2.go

```diff
@@ -0,0 +1,195 @@
+package v2
+
+import (
+	"context"
+	"errors"
+	logContext "github.com/trufflesecurity/trufflehog/v3/pkg/context"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors/azure_entra"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors/azure_entra/serviceprincipal"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+	detectors.DefaultMultiPartCredentialProvider
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ interface {
+	detectors.Detector
+	detectors.Versioner
+} = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	SecretPat = regexp.MustCompile(`(?:[^a-zA-Z0-9_~.-]|\A)([a-zA-Z0-9_~.-]{3}\dQ~[a-zA-Z0-9_~.-]{31,34})(?:[^a-zA-Z0-9_~.-]|\z)`)
+)
+
+func (s Scanner) Version() int {
+	return 2
+}
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"q~"}
+}
+
+// FromData will find and optionally verify Azure secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	clientSecrets := findSecretMatches(dataStr)
+	if len(clientSecrets) == 0 {
+		return results, nil
+	}
+	clientIds := azure_entra.FindClientIdMatches(dataStr)
+	tenantIds := azure_entra.FindTenantIdMatches(dataStr)
+
+	client := s.client
+	if client == nil {
+		client = defaultClient
+	}
+
+	processedResults := ProcessData(ctx, clientSecrets, clientIds, tenantIds, verify, client)
+	for _, result := range processedResults {
```

Comment:
suggestion: Same as above, we can flatten this for loop.
`results = append(results, processedResults...)`